### DISCUSSION TITLE
Use Azure Service Bus for message queue?

### DISCUSSION THREAD

**Discussion by spmsh** [miguelgrinberg/Flask-SocketIO#2047] [Q&A] [answered]
Has anyone used Azure services (Azure Service Bus, Azure Event Grid, or else) for message queue, instead of Redis/RabbitMQ, and if so, what would the connection look like when doing `socketio = SocketIO(app, message_queue=....)` what to put in the message_queue param here? Thanks.

**Comment by miguelgrinberg** [comment, marked as answer]
You would need to write a custom subclass of the `PubSubManager` class in package python-socketio. Take a look at the Redis implementation, for example: https://github.com/miguelgrinberg/python-socketio/blob/main/src/socketio/redis_manager.py.

Once you write a similar class for your queue, you can pass an instance of it to the `message_queue` argument for Socket.IO to use.
